src/orion/algo/space/__init__.py

```diff
@@ -1095,6 +1097,43 @@ def __setitem__(self, key, value):
             )
         super().__setitem__(key, value)
 
+    def assert_contains(self, trial):
+        """Same as __contains__ but instead of return true or false it will raise an exception
+        with the exact causes of the mismatch.
+
+        Raises
+        ------
+        ValueError is the trial has parameters that are not contained by the space.
+
+        """
+        if isinstance(trial, str):
+            return super().__contains__(trial)
+
+        flattened_params = flatten(trial.params)
+        keys = set(flattened_params.keys())
+        errors = []
+
+        for dim_name, dim in self.items():
+            if dim_name not in keys:
+                errors.append(f"{dim_name} is missing")
+                continue
+
+            value = flattened_params[dim_name]
+            if value not in dim:
+                errors.append(f"{value} does not belong to the dimension {dim}")
+
+            keys.remove(dim_name)
+
+        if len(errors) > 0:
+            errors = "\n - ".join(errors)
+            raise ValueError(f"Trial {trial.id} is not contained in space:\n{errors}")
+
+        if len(keys) != 0:
+            errors = "\n - ".join(keys)
+            raise ValueError(f"Trial {trial.id} has additional parameters:\n{errors}")
+
+        return True
+
     def __contains__(self, key_or_trial):
```

Comment:
Seems like there is a lot of duplication. Could __contains__ simply call assert_contains, try-catch and return True if successful otherwise False?

Reply:
done